enumerable.rb

```diff
@@ -0,0 +1,205 @@
+        result = true if yield(item) == true
+      end
+    elsif param.nil? == false
+      if param.instance_of?(Class)
+        to_a.my_each do |item|
+          result = true if item.is_a?(param) == true
+        end
+      elsif param.instance_of?(Regexp)
+        to_a.my_each do |item|
+          result = true if param.match?(item) == true
+        end
+      else
+        to_a.my_each do |item|
+          result = true if item == param
+        end
+      end
+    else
+      to_a.my_each do |item|
+        result = true unless item.nil? || item == false
+      end
+    end
+    result
+  end
+
+  def my_none?(param = nil)
+    result = true
+    if block_given?
+      to_a.my_each do |item|
+        result = false if yield(item) == true
+      end
+    elsif param.nil? == false
+      if param.instance_of?(Class)
+        to_a.my_each do |item|
+          result = false if item.is_a?(param) == true
+        end
+      elsif param.instance_of?(Regexp)
+        to_a.my_each do |item|
+          result = false if param.match?(item) == true
+        end
+      else
+        to_a.my_each do |item|
+          result = false if item == param
+        end
+      end
+    else
+      to_a.my_each do |item|
+        result = false unless item.nil? || item == false
+      end
+    end
+    result
+  end
+
+  def my_count?(param = nil)
+    count = 0
+    if block_given?
+      to_a.my_each do |item|
+        count += 1 if yield(item) == true
+      end
+    elsif param.nil? == false
+      to_a.my_each do |item|
+        count += 1 if item == param
+      end
+    else
+      to_a.my_each do
+        count += 1
+      end
+    end
+    result
+  end
+
+  def my_map(param = nil)
+    return to_enum(:my_map, param) unless !param.nil? || block_given?
+
+    result_array = []
+
+    if param
+      to_a.my_each do |item|
+        result_array.push(param.call(item))
+      end
+      result_array
+
+    elsif block_given?
+      to_a.my_each do |item|
+        result_array.push(yield(item))
+      end
+      result_array
+    end
+  end
+
+  def my_inject(param_1 = nil, param_2 = nil)
+    array = to_a
+
+    if param_1 && param_2
+      acumulator = param_1
+      operator = param_2
+
+      array.my_each do |item|
+        acumulator = acumulator.send(operator, item)
+      end
+      acumulator
+
+    elsif param_1 && param_2.nil? && !block_given?
+      acumulator = array.shift
+      operator = param_1
+
+      array.my_each do |item|
+        acumulator = acumulator.send(operator, item)
+      end
+      acumulator
+
+    elsif block_given? && param_1
+      acumulator = param_1
+
+      array.my_each do |item|
+        acumulator = yield(acumulator, item)
+      end
+      acumulator
+
+    elsif block_given?
+      acumulator = array.shift
+
+      array.my_each do |item|
+        acumulator = yield(acumulator, item)
+      end
+      acumulator
+
+    else
+      raise StandardError, 'wrong use'
+    end
+  end
+end
```

Comment:
- [ ] **my_inject should raise a "LocalJumpError" when no block or argument is given.** Create a random array `arr = [1,2,3]` and test the original `my_inject` without giving it any block -> `arr.inject`. This will give a LocalJumpError because the inject method expects some sort of valid input. Your `my_inject` should do the same.

![image](https://user-images.githubusercontent.com/64360974/115000828-ce241700-9eab-11eb-8500-f0cd4e6d76b2.png)



- [ ] **my_inject should not mutate the original array.** Check the screenshot below. It's mutating the array. 

![image](https://user-images.githubusercontent.com/64360974/115001982-e0528500-9eac-11eb-888f-692bc27863ef.png)



- [ ] **my_inject when a symbol is specified without an initial value combines each element of the collection by applying the symbol as a named method.**

- While your `my_inject` is giving the correct output, it's mutating the array to [2,3,4,5, 6] hence the original inject method is giving 20. This issue will be solved when you fix the previously stated bug.

![image](https://user-images.githubusercontent.com/64360974/115002471-74bce780-9ead-11eb-9342-6ba84ec28395.png)